ros/angel_system_nodes/angel_system_nodes/intent_to_command.py

```diff
@@ -0,0 +1,134 @@
+        )
+        self._sys_cmd_topic = (
+            self.declare_parameter("sys_cmd_topic", "")
+            .get_parameter_value()
+            .string_value
+        )
+        self._intent_to_cmd_map_path = (
+            self.declare_parameter("intent_to_cmd_map", "")
+            .get_parameter_value()
+            .string_value
+        )
+
+        if self._explicit_intent_topic == "":
+            raise ValueError(
+                "Please provide the explicit intent topic with the `explicit_intent_topic` parameter"
+            )
+        if self._confirmed_intent_topic == "":
+            raise ValueError(
+                "Please provide the confirmed intent topic with the `confirmed_intent_topic` parameter"
+            )
+        if self._intent_to_cmd_map_path == "":
+            raise ValueError(
+                "Please provide the intent command map path with the `intent_to_cmd_map` parameter"
+            )
+        if self._sys_cmd_topic == "":
+            raise ValueError(
+                "Please provide the system command topic with the `sys_cmd_topic` parameter"
+            )
+
+        log = self.get_logger()
+        log.info(f"Explicit intent topic: {self._explicit_intent_topic}")
+        log.info(f"Confirmed intent topic: {self._confirmed_intent_topic}")
+        log.info(f"System command topic: {self._sys_cmd_topic}")
+        log.info(f"Intent to cmd map: {self._intent_to_cmd_map_path}")
+
+        # Load intent to cmd map (yaml)
+        with open(self._intent_to_cmd_map_path, 'r') as f:
+            config = yaml.safe_load(f)
+        self._intent_to_cmd_map = config["commands"]
+
+        self._subscription = self.create_subscription(
+            InterpretedAudioUserIntent,
+            self._explicit_intent_topic,
+            self.intent_callback,
+            1
+        )
+        self._subscription = self.create_subscription(
+            InterpretedAudioUserIntent,
+            self._confirmed_intent_topic,
+            self.intent_callback,
+            1
+        )
+        self._sys_cmd_publisher = self.create_publisher(
+            SystemCommands,
+            self._sys_cmd_topic,
+            1
+        )
+
+    def intent_callback(self, intent: InterpretedAudioUserIntent) -> None:
+        """
+        Callback function for the user intent messages. Converts the intent
+        to a system command and publishes the command to the system command
+        topic.
+        """
+        log = self.get_logger()
+
+        try:
+            sys_cmd = self._intent_to_cmd_map[intent.user_intent]
+        except KeyError:
+            log.warning(
+                f"User intent `{intent.user_intent}` not found in sys cmd map. "
+                f"Available intents are {list(self._intent_to_cmd_map.keys())}"
+            )
+            return
+
+        sys_cmd_msg = SystemCommands()
+
+        try:
+            # Set the corresponding sys cmd field to True
+            setattr(sys_cmd_msg, sys_cmd, True)
```

Comment:
This is OK for now since we just have a few boolean-based commands, but this will break down when we have anything more complex. I'm not sure I have a concrete recommendation other than adding a comment before this that acknowledges this.